### PR TITLE
Add daily test script and analysis python script

### PR DESCRIPTION
**Pull request by xuyangit**

### CODE REVIEW

File: oap_perf_test.sh [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+workDir=`cd ${BASH_SOURCE%/*}; pwd`
+mailList=(yang2.xu@intel.com yue.a.wang@intel.com carson.wang@intel.com hao.cheng@intel.com daoyuan.wang@intel.com linhong.liu@intel.com lei.l.li@intel.com chenzhao.guo@intel.com yi.cui@intel.com)
+# source environment variables and check requirements
+function sourceEnv {
+	[ -f $OAP_HOME/.bash_profile ] && . $OAP_HOME/.bash_profile
+	[ -f $OAP_HOME/.bash_login ] && . $OAP_HOME/.bash_login
+	[ -f $OAP_HOME/.profile ] && . $OAP_HOME/.profile
+	[ -f $OAP_HOME/.bashrc ] && . $OAP_HOME/.bashrc
+	if ! type git >/dev/null 2>&1; then
+		echo "Git not installed, pls install it first!" >&3 && return 1
```
Comment (yueawang) [inline oap_perf_test.sh]: Please indent the script in a consistent way, at least tab => 4 space.
Also, code style should be kept in a consistent way.

File: oap_perf_test.sh [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+	scriptPath=${workDir}/${scriptName} 
+	if ! crontab -l 2>&1 | grep -q "$scriptName"; then
+		if crontab -l 2>&1 | grep -q "no crontab for"; then
+			echo "30 2 * * * ${scriptPath} -c" | crontab -
+		else 
+			(crontab -l; echo "30 2 * * * ${scriptPath} -c") | crontab -
+		fi
+	fi
+}
+function cloneAndBuild {
```
Comment (yueawang) [inline oap_perf_test.sh]: some functions have space & others not.

File: oap_perf_test.sh [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+	--class org.apache.spark.sql.OapPerfSuite \
+	${workDir}/oap-perf-suite/target/scala-2.11/oap-perf-suite-assembly-1.0.jar \
+	-d
+}
+# create task directory and run periodic task
+function runTask {
+	cloneAndBuild
+	if [ "$?" -ne 0 ]; then return 1; fi
+	# copy hive database meta file to current directory
+	cd ${workDir}/${resDir} && cp -r ../metastore_db ./
```
Comment (yueawang) [inline oap_perf_test.sh]: Is this metastore_db created during genData??

File: oap_perf_test.sh [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+      lastPath=${workDir}/last_test_info                                         
+      [ -f $lastPath ] && . $lastPath                                            
+      todayPath=${workDir}/${resDir}/testres_${today}                            
+      if [ -n "$lastResPath" ]; then                                             
+        python ${workDir}/analyze.py $lastResPath $todayPath ${workDir}/${resDir}/cmp.html
+      else                                                                       
+        python ${workDir}/analyze.py $todayPath $todayPath ${workDir}/${resDir}/cmp.html
+      fi                                                                         
+      echo "lastResPath=${todayPath}" > $lastPath                                
+    fi
```
Comment (yueawang) [inline oap_perf_test.sh]: A few lines end with space, please clean them up.

File: analyze.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
+import sys
+import os
+def parseFile(fileName):
+    file = open(fileName)
```
Comment (yueawang) [inline analyze.py]: what if open fail?

File: analyze.py [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    res = {}
+    for i, line in enumerate(file.readlines()):
+        totalLines += 1
+        if(line.startswith("#")):
+            suitesLines.append(i + 1)
+            suitesNames.append(line)
+    suitesLines.append(totalLines)
+    file.seek(0)
+    lines = file.readlines()
+    for i in range(len(suitesNames)):
```
Comment (yueawang) [inline analyze.py]: Why not finish parse in one loop??

File: analyze.py [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+            tpres[caseName] = {}
+            while(lines[suiteStart - 1].startswith("|")):
+                median = int(lines[suiteStart - 1].split("|")[-2].lstrip())
+                configName = lines[suiteStart - 1].split("|")[1].lstrip()
+                suiteStart += 1
+                tpres[caseName][configName] = median
+        res[suitesNames[i]] = tpres
+    return res
+
+style="""<style type="text/css">
```
Comment (yueawang) [inline analyze.py]: Useless??

File: analyze.py [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+                    else:
+                        suiteTable += """<td class="tg-lqy6">{}</td>\n""".format("N/A")
+                        suiteTable += """<td class="tg-yw4l">{}</td>\n""".format(median)
+                        suiteTable += """<td class="tg-yw4l">N/A</td>\n"""
+                    suiteTable += "</tr>\n"
+                    j += 1
+            suiteTable += "</table>\n"
+            htmlTables += "\n{}\n".format(suiteTable)
+    cmpFile=open(args[3], mode='w')
+    cmpFile.write(htmlContent + htmlTables + "</body>\n</html>")
```
Comment (yueawang) [inline analyze.py]: Does move html tag "< /body > \n < /html >" to htmlContent have better looking??

**Review by yueawang — COMMENTED** [review]
A few comments on coding style and logic.

File: oap_perf_test.sh (hunk 1 above)
Comment (xuyangit) [reply oap_perf_test.sh]: It looks well on shell but don't know why changes here.  Will change to 4 space

File: oap_perf_test.sh (hunk 2 above)
Comment (xuyangit) [reply oap_perf_test.sh]: @yueawang ,  dont get ur point as all look the same.

File: oap_perf_test.sh (hunk 3 above)
Comment (xuyangit) [reply oap_perf_test.sh]: yes, for the first time we run this script with -g, i will update the script later

File: oap_perf_test.sh (hunk 2 above)
Comment (xuyangit) [reply oap_perf_test.sh]: @yueawang ,u mean new line?

Comment (yueawang) [reply oap_perf_test.sh]: Yes

File: oap_perf_test.sh (hunk 1 above)
Comment (yueawang) [reply oap_perf_test.sh]: That's because you use tab

File: oap_perf_test.sh (hunk 4 above)
Comment (xuyangit) [reply oap_perf_test.sh]: @yueawang , together fixed with above issue

File: oap_perf_test.sh (hunk 1 above)
Comment (yueawang) [reply oap_perf_test.sh]: Have you fix all tab??

File: analyze.py (hunk 7 above)
Comment (yueawang) [reply analyze.py]: What is this style used for??

File: analyze.py (hunk 5 above)
Comment (xuyangit) [reply analyze.py]: Python will fail with exit code 1, maybe we can add checking in script, if analysis fails, we mail the reason.

File: analyze.py (hunk 6 above)
Comment (xuyangit) [reply analyze.py]: Try in one loop but not convenient to collect result.

**Review by yueawang — COMMENTED** [review]
LGTM, can you create a folder (e.g. scripts) and put these files into it??

**Comment by xuyangit** [comment]
@yueawang ,pls help review,tks

**Pull request merged**